task3/checkpoint_fsync.c

```diff
@@ -78,6 +82,13 @@ init_matrix (double** matrix)
 	}
 }
 
+inline long nano_diff(long l, long r) {
```

Comment:
This is only called at two points, the `l > r` will always be false in these cases. 
I would suggest to let this take the `timespec` structs and compute the counter increase here, will make it more readable.

Reply:
The `l < r` condition can be met for operations taking longer than one second. Which could happen in file synchronization, just a possibility I didn't want to forego.
You're right with the timespec, we can encapsulate some non-nice code there from our calculation function. Will move.